pkg/master/controller.go

```diff
@@ -268,13 +260,10 @@ func (c *Controller) CreateOrUpdateMasterServiceIfNeeded(serviceName string, ser
 			Type:            serviceType,
 		},
 	}
-	if err := rest.BeforeCreate(service.Strategy, ctx, svc); err != nil {
-		return err
-	}
 
 	_, err := c.ServiceClient.Services(api.NamespaceDefault).Create(svc)
-	if err != nil && errors.IsAlreadyExists(err) {
-		err = nil
+	if errors.IsAlreadyExists(err) {
+		return c.CreateOrUpdateMasterServiceIfNeeded(serviceName, serviceIP, servicePorts, serviceType, reconcile)
```

Comment:
> I don't understand this change - can you please clarify? Wouldn't this be an infinite loop?

The next time we call in, we should take the first `if` since the `Get` is successful.  Even if we hit a lagging follower, we'll eventually free up and the reconcile logic that should run the first time will run.